### ISSUE TITLE
Compatibility Issues with Python 3.9+ and XGBoost Version Mismatch

### ISSUE THREAD

**Issue by roei-shlezinger** [amir-zeldes/RFTokenizer#16]
Hi,  

I encountered some issues related to Python version compatibility and dependency conflicts while using the package:  

1. Running the package with Python 3.9+ results in the following error:  
   ```
   AttributeError: module 'collections' has no attribute 'Mapping'
   ```  
   This appears to be due to changes in the `collections` module in Python 3.9.  

2. Downgrading to Python 3.8 to resolve this issue led to an error related to an XGBoost version mismatch when loading a serialized model:  
   ```
   xgboost.core.XGBoostError: [...] If you are loading a serialized model (like pickle in Python, RDS in R) or configuration generated by an older version of XGBoost, please export the model by calling `Booster.save_model` from that version first, then load it back in current version.
   ```  

Would it be possible to:  
1. Update the `requirements.txt` file to include a compatible XGBoost version?  
2. Specify the supported Python version in the documentation or as part of the package configuration?  

This would help clarify the setup process and prevent similar issues for other users.  

Thank you so much for your time and for maintaining this package!

**Comment by amir-zeldes** [comment]
Hm, that's odd, I can't reproduce this behavior, at least not with Python 3.12 on a fresh clone.

```
> python --version
Python 3.12.2

> cd rftokenizer
> python tokenize_rf.py -m heb example_in.txt
! Model file C:\Uni\CL\RFTokenizer\_test\rftokenizer\models\heb.seg not found
! Attempting to download it... (this could take a while)
! Done!

עיפרון
עיפרון
הוא
כלי
כתיבה
ידני
לשם
כתיבה
ו|ציור
...
```

Looking at the library versions this is running under xgboost==2.0.3 and flair==0.13.0, exactly as specified in the requirements. I also don't think RFTokenizer uses Mapping anywhere:

https://github.com/search?q=repo%3Aamir-zeldes%2FRFTokenizer%20Mapping&type=code

So I suspect one of your imports has an incompatible version and is triggering the warning, but I can't tell which. Can you post the entire trace to show what is attempting to import Mapping?

**Comment by roei-shlezinger** [comment]
I made a Colab notebook, check it out, the bug shows up there when i install the package (using pip and git clone):
https://colab.research.google.com/drive/1ko5bCDS_gMRkYjrg14T6hqiRs-PMxTop?usp=sharing

This installation bug seems to be recurring, and I was wondering if you've looked into tools that can help manage dependencies more easily? Using a package manager like Poetry could make keeping track of dependencies more straightforward, and might prevent these kinds of issues from popping up. It might be worth considering if you haven't already

**Comment by amir-zeldes** [comment]
Thanks for sharing your notebook - it looks like python 3.10 there, but that shouldn't matter. Actually looking at the error more closely, this is not a version error for the dependencies per se, at least not in the code - it's just an outdated model. The reason I wasn't able to reproduce it with the command above is that I ran the Hebrew model, which was updated more recently, but it looks like the distributed Arabic model was trained on an older version of XGBoost.

I can train a fresh model and package a new release.
